aps2scala/dump-scala.cc

```diff
@@ -92,6 +94,35 @@ static string program_id(string name)
   return result;
 }
 
+/**
+ * Utility function to dump static circular evaluation trait to
+ * by pass some of CircularEvaluation limitations
```

Comment:
What's wrong with CircularEvaluation?  Maybe it should be fixed, not just extended?

Reply:
Added a description of _what would go wrong with_ `CircularEvaluation` **IF** we use it for static circular evaluation.

More specifically, `CircularEvaluation` doesn't record a `change` in attribute instance value and does not like attribute value being re-assigned.